state/executor.go

```diff
@@ -132,10 +132,6 @@ func (e *BlockExecutor) CreateBlock(height uint64, lastCommit *types.Commit, las
 
 // ApplyBlock validates and executes the block.
 func (e *BlockExecutor) ApplyBlock(ctx context.Context, state types.State, block *types.Block) (types.State, *cmstate.ABCIResponses, error) {
-	err := e.Validate(state, block)
```

Comment:
i don't think you need to remove this. the test should still pass. if you remove this and call `Validate` separately, it is fine while syncing the block as you call `Validate` first before `applyBlock`, however while publishBlock, you skip the validation and directly `applyBlock` and then do the validation. I guess this is fine, but no-harm doing double validation either.

Reply:
Restored.